### DISCUSSION TITLE
%avi encoder seems to be broken

### DISCUSSION THREAD

**Discussion by Rafi594** [savonet/liquidsoap#3307] [User Support]
Hello,

I'm currently trying to use output.external for using custom ffmpeg command but the output have a blinking green screen (probably pixel format issues)

I have also tried with `output.file(%avi, fallible=true, "video.flv", stream)` and I get the same issue.

I made a workaround by replacing %avi by `%ffmpeg(format="avi",%video(codec="rawvideo", pixel_format="yuv420p"),%audio(codec="pcm_s16le"))`

Script for reproducing
```
audio = input.http("http://127.0.0.1:8000/audio.aac")

video = video.testsrc.ffmpeg(pattern="testsrc2")

stream = mux_video(video=video, audio)

output.external(
  self_sync=true,
  fallible=true,
  %avi,
  "ffmpeg -re -i pipe:0 -f avi -c:v h264_nvenc -b:v 5000k -c:a aac -b:a 128k -pix_fmt yuv420p -f flv rtmp://127.0.0.1:1935/test-stream",
  stream)
```

https://github.com/savonet/liquidsoap/assets/28735472/79f7b57f-5203-48b8-94fc-52aa9206492f

**Comment by toots** [comment]
Thanks for reporting, glad you found a workaround.
